### ISSUE TITLE
`strategy: prefix` doesn't work when locales are set in `i18n.config.ts` and fetched from API but not hard coded in `nuxt.config.ts`

### ISSUE THREAD

**Issue by serhii-chernenko** [nuxt-modules/i18n#3288]
### Environment

- Operating System: Darwin
- Node Version:     v20.14.0
- Nuxt Version:     3.15.0
- CLI Version:      3.17.2
- Nitro Version:    2.10.4
- Package Manager:  npm@10.9.2
- Builder:          -
- User Config:      default
- Runtime Modules:  @nuxtjs/i18n@9.0.0-rc.2
- Build Modules:    -

### Reproduction

The repo is available here: https://github.com/serhii-chernenko/nuxt-i18n-issue

I also added files to the Stackblitz service: https://stackblitz.com/edit/nuxt-starter-e83jtjvk?file=nuxt.config.ts, but it just doesn't work with dynamic routes. It'd be better to reproduce locally from the GitHub repo mentioned above.

### Describe the bug

# Nuxt i18n dynamic locales issue explanation

The main goal I want to achieve is fetching locales' setup from a DB.

I have the languages API endpoint [`languages.get.ts`](https://github.com/serhii-chernenko/nuxt-i18n-issue/blob/main/server/api/languages.get.ts):
```ts
export default defineEventHandler(() => {
  return {
    fallbackLocale: 'en',
    locales: [
        { code: 'en', name: 'English' },
        { code: 'ua', name: 'Ukrainian' }
    ],
    messages: {
        en: {
          home: 'Home',
          about: 'About Us'
        },
        ua: {
          home: 'Головна',
          about: 'Про нас'
        }
    },
  };
});
```

The data is fetched in [`i18n.config.ts`](https://github.com/serhii-chernenko/nuxt-i18n-issue/blob/main/i18n.config.ts):
```ts
export default defineI18nConfig(async () => {
    return await $fetch('/api/languages');
})
```

The file is imported via [`nuxt.config.ts`](https://github.com/serhii-chernenko/nuxt-i18n-issue/blob/main/nuxt.config.ts):
```ts
export default defineNuxtConfig({
  i18n: {
    strategy: 'prefix',
    defaultLocale: 'en',
    vueI18n: './i18n.config.ts',
    // locales: ['en', 'ua'],
  }
})
```

As a result, translations work as expected, but `strategy: prefix` is not. `en` and `ua` locale codes are not included in the URL. 
![image](https://github.com/user-attachments/assets/7832c9ea-d94b-40ea-a565-59c7ea80206b)
![image](https://github.com/user-attachments/assets/64414a9e-d05a-4d67-9893-b939019f6256)

The only way I found to fix it, just add:
```ts
locales: ['en', 'ua'],
```
to the [`nuxt.config.ts`](https://github.com/serhii-chernenko/nuxt-i18n-issue/blob/main/nuxt.config.ts#L13):
![image](https://github.com/user-attachments/assets/1cd04086-908c-491a-897b-6dd12d6dbfca)
![image](https://github.com/user-attachments/assets/0ba93dee-d130-4717-9350-e986ea0602d4)

But it doesn't make sense because it's still hardcoded anyway. I want to make `strategy: prefix` working when the whole functionality is dynamic and fetched from the API endpoint (from DB in general).

### Logs

```shell

```

**Comment by BobbieGoede** [comment]
This is expected behavior, route localization happens at build-time while the `i18n.config` is loaded at runtime.

Dynamic/runtime routing is often requested, but technically not straightforward (especially so when taking perf into consideration).

**Comment by serhii-chernenko** [comment]
@BobbieGoede do you have any ideas on how I can solve this until the feature is implemented in the module? I suspect, I can write a custom module or just a plugin with a build hook to fetch locales from a DB directly and provide them to the nuxt config. Is it a right direction?

**Comment by BobbieGoede** [comment]
Maybe I misunderstood the behavior you're trying to achieve, do you want the config to be fetched and applied at build-time (you could implement this) or do you want the config to change at runtime (this won't work currently)?

**Comment by maxdzin** [comment]
@BobbieGoede, maybe `serverRoutePrefix` could be used to achieve this?

**Comment by serhii-chernenko** [comment]
@BobbieGoede no, only build time is good. At least, I thought about only build-time hooks when I wrote this
> I suspect, I can write a custom module or just a plugin with a build hook to fetch locales from a DB directly and provide them to the nuxt config. Is it a right direction?

**Comment by BobbieGoede** [comment]
Locales are build-time configuration, so fetching them from an API at runtime is not currently supported. Closing as duplicate of #2289, which tracks runtime locale configuration.
